analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxFacade.cs

```diff
@@ -53,6 +54,16 @@ internal sealed class CSharpSyntaxFacade : SyntaxFacade<SyntaxKind>
 
         public override bool IsNullLiteral(SyntaxNode node) => node.IsNullLiteral();
 
+        public override IEnumerable<SyntaxNode> ArgumentExpressions(SyntaxNode node) =>
+            node switch
+            {
+                ObjectCreationExpressionSyntax objectCreation => objectCreation.ArgumentList?.Arguments.Select(x => x.Expression),
+                null => Enumerable.Empty<SyntaxNode>(),
+                _ => ImplicitObjectCreationExpressionSyntaxWrapper.IsInstance(node)
+                ? Enumerable.Empty<SyntaxNode>()
+                : throw InvalidOperation(node, nameof(ArgumentExpressions))
```

Comment:
We can do it like this: https://github.com/SonarSource/sonar-dotnet/blob/4694178cfc3a796c7bb64e406cf3ce3a9f462fe2/analyzers/src/SonarAnalyzer.CFG/Extensions/IOperationExtensions.cs#L110-L117
```suggestion
                var _ when ImplicitObjectCreationExpressionSyntaxWrapper.IsInstance(node) => Enumerable.Empty<SyntaxNode>(),
                _ => throw InvalidOperation(node, nameof(ArgumentExpressions))
```